### ISSUE TITLE
Back Button in Multiple Views Deletes Shared Controller

### ISSUE THREAD

**Issue by seljad** [jonataslaw/getx#2888]
I have an app that contains a bottom-navigation bar with nested routing, utilizing the GetX ecosystem. Some of the views have the same controllers. When a user is in one of these views and clicks the back button, Get.back() is executed, resulting in the removal of both the view and the controller from memory.

My issue arises when I click on the back button of the first created view. The possible scenario is as follows: a user opens page A, which is the first item in the bottom navigation bar. GetX creates a controller for page A. If the user then moves to the second item in the bottom navigation bar and attempts to recreate view A, the view is indeed created and used the previously mentioned controller in memory.

Now, a problem occurs when the user clicks on the back button in the second item of bottom-navigation bar. This action leads to the removal of view A and the subsequent deletion of the controller from memory, despite another view with the same controller being connected to it.

I am using GetX version 4.6.5

![autodraw 8_26_2023](https://github.com/jonataslaw/getx/assets/29150778/6451bec1-43e2-4fd1-bc4b-e9a55562383b)

**Comment by jonataslaw** [comment]
Are you using onGenerateRoute and Navigator widget?

**Comment by seljad** [comment]
> Are you using onGenerateRoute and Navigator widget?

@jonataslaw 
look at this repo and video:
https://github.com/seljad/getx-nesting-routes

**Comment by jonataslaw** [comment]
Thank you!
let me have a look at this

**Comment by seljad** [comment]
> Thank you!
> let me have a look at this

I'm waiting for your response

**Comment by seljad** [comment]
@jonataslaw 

any idea?

**Comment by seljad** [comment]
can you help me with that, please? @jonataslaw

**Comment by seljad** [comment]
> Hi @seljad, Do you find any solution?

No, It's too bad that something like this happens, And I am waiting for @jonataslaw to answer that.

**Comment by jonataslaw** [comment]
How do you create the controller? Get.put in the view, Binding, or GetBuilder widget?

**Comment by seljad** [comment]
> How do you create the controller? Get.put in the view, Binding, or GetBuilder widget?

I have shared with you my project [in here](https://github.com/seljad/getx-nesting-routes). as you see, I created a controller in binding with `Get.lazyPut<HomeController>(
      () => HomeController(),
    );`

**Comment by seljad** [comment]
@jonataslaw any idea?

**Comment by seljad** [comment]
@jonataslaw is this fix in version 5?

**Comment by wildsurfer** [comment]
@jonataslaw facing the same problem
